### PR TITLE
Feat/date time picker ts

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
This PR is a little big, but there's nothing exciting about it.. this is what I did
- Covert BaseInput file to TS and set `// @ts-nocheck` to avoid TS checking
- Convert (old) TextField file to TS and also set `// @ts-nocheck` to avoid TS checking
- Migrated DateTimePicker component to TS 
- Fix minor relevant TS with Dialog's `height` prop 

## Changelog
Convert DateTimePicker component to TS

### CODE REVIEW

File: src/components/dateTimePicker/index.tsx [hunk 1]
```diff
@@ -237,7 +243,7 @@ class DateTimePicker extends Component {
           iconStyle={{tintColor: Colors.grey10}}
           onPress={this.toggleExpandableOverlay}
         />
-        <Button link iconSource={Assets.icons.check} useCustomTheme={useCustomTheme} onPress={this.onDonePressed}/>
+        <Button link iconSource={Assets.icons.check} useCustomTheme onPress={this.onDonePressed}/>
```
Comment (M-i-k-e-l) [inline src/components/dateTimePicker/index.tsx:246]: Are you sure we want to pass `useCustomTheme` as true? Don't we want to ignore the ts error?

File: src/components/dateTimePicker/index.tsx [hunk 2]
```diff
@@ -37,87 +28,102 @@ const THEME_VARIANTS = {
...
+   */
+  headerStyle?: StyleProp<ViewStyle>;
+  /**
+   * Render custom input
+   */
+  renderInput?: () => React.ReactElement;
+  /**
+   * Override system theme variant (dark or light mode) used by the date picker.
+   */
+  themeVariant?: 'light' | 'dark';
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/dateTimePicker/index.tsx:101]: It's not our variant. 
It's a type coming from DateTimePicker library

File: src/components/baseInput/Validators.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import _ from 'lodash';
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/baseInput/Validators.ts:1]: I don't think it's such an important file

File: src/components/dateTimePicker/index.tsx [hunk 4]
```diff
@@ -1,31 +1,22 @@
...
-import {TextField} from '../inputs';
-import Dialog from '../dialog';
+import {asBaseComponent, BaseComponentInjectedProps} from '../../commons/new';
+import TextField from '../textField';
+import Dialog, {DialogProps} from '../dialog';
 import View from '../view';
 import Button from '../button';
 
-
 const MODES = {
```
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/dateTimePicker/index.tsx:15]: Fixed. 
I moved the example screen to ts
I fixed the export in generatedTypes/index
and fix various errors in the example screen

File: src/components/dateTimePicker/index.tsx (hunk 1 above)
Comment (ethanshar) [reply src/components/dateTimePicker/index.tsx]: Im assuming it will always need to apply useCustomTheme 
But I see your point, added it back and ignored

**Review by M-i-k-e-l — APPROVED** [review]
Approving, you can switch `MODES` to `type MODES = 'date' | 'time';` if you like.

**Pull request merged**